components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c

```diff
@@ -134,24 +140,57 @@ static void psa_mac_operation(void)
                 }
 
                 case PSA_MAC_UPDATE: {
-                    uint8_t *input_ptr = mbedtls_calloc(1, msg.in_size[1]);
-                    if (input_ptr == NULL) {
+
+                    uint8_t * input_buffer = NULL;
+                    size_t data_remaining = msg.in_size[1];
+                    size_t allocation_size = data_remaining;
+                    size_t size_to_read = 0;
+                    psa_status_t failure_status = PSA_SUCCESS;
+                    
+                    if (allocation_size > MAX_DATA_CHUNK_SIZE)
+                    {
+                        allocation_size = MAX_DATA_CHUNK_SIZE;
+                    }
+
+                    input_buffer = mbedtls_calloc(1, allocation_size);
+                    if (input_buffer == NULL) {
```

Comment:
i see the entire block from uint8_t * input_buffer = NULL;
and until here repeats for every similar operation.
how about adding internal static service chunk alloc function and calling
it from every such scenario, instead of duplicating code

Reply:
wrapping the processing in chunks loop is a bit problematic as at the core of the loop are calls to two functions (psa_hash_update and psa_mac_update) which are very similar, but have a different parameter signature. this means we need to resort to some ugly tricks to have the helper function handle both, an I don't think this is justified.